### PR TITLE
Babashka compatibility

### PR DESCRIPTION
**Pull request by borkdude**
This PR adds babashka compatibility to deep-diff2

### CODE REVIEW

**Comment by borkdude** [comment]
This needs to be executed with bb from master for now:

```
bash <(curl https://raw.githubusercontent.com/babashka/babashka/master/install) --version 1.0.166-SNAPSHOT --dir .
./bb test:bb
```

Not sure how to add this to the CircleCI config.

Also this depends on a git dep of clj-diff, so we might want to wait for a mvn release of that first.

File: bb.edn [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+  test:bb {:doc "Run babashka tests with custom runner"
+           :extra-paths ["src" "test"]
+           :extra-deps {current/project {:local/root "."}
+                        org.clojure/test.check    {:mvn/version "1.1.1"}}
+           :task (exec 'lambdaisland.deep-diff2.runner/run-tests)}
+  npm-ws (when-not (fs/exists? "node_modules/ws")
+           (shell "npm install ws"))
+  test:kaocha {:doc "Run kaocha tests"
+               :depends [npm-ws]
+               :task (apply clojure "-M:dev:test -m kaocha.runner" *command-line-args*)}}}
```
Comment (plexus) [inline bb.edn:15]: I would appreciate not adding this to all of our projects, the standard way to run kaocha is through `bin/kaocha`.

Comment (borkdude) [reply bb.edn]: Done

**Comment by borkdude** [comment]
The updated PR should now work with the latest stable release of bb.

I have looked into how to test this in CI, but I'm not familiar enough with the orb / templating stuff. Any pointers would be appreciated.

**Comment by plexus** [comment]
yeah it's convoluted... FWIW the orb sources are here https://github.com/lambdaisland/open-source/tree/main/circleci

I'd be ok with having a github action for the bb tests, and still running the kaocha tests on circle, since that's what we still default to for most of our projects... I don't particularly like either of them which also means I don't have a particular preference :)

**Comment by borkdude** [comment]
@plexus I'll add a github config for bb tests

**Comment by borkdude** [comment]
Somehow it's not finding lambdaisland.deepdiff-2.runner in the Github Action, while it works correctly for me local (on an M1 Air). But when I use [act](https://github.com/nektos/act) to run the Action locally, I can reproduce what happens in CI... investigating.

**Comment by borkdude** [comment]
@plexus Ready for review. You can see the Github tests running here: https://github.com/borkdude/lambdaisland-deep-diff2/tree/babashka

**Comment by plexus** [comment]
Fabulous!

released in

```clj
[lambdaisland/deep-diff2 "2.6.166"]
{lambdaisland/deep-diff2 {:mvn/version "2.6.166"}}
```

**Pull request merged**